addons/l10n_it_edi/models/res_partner.py

```diff
@@ -26,3 +29,21 @@ class ResPartner(models.Model):
             "CHECK(l10n_it_pa_index IS NULL OR LENGTH(l10n_it_pa_index) >= 6)",
             "PA index must have between 6 and 7 characters."),
     ]
+
+    @api.model
+    def _l10n_it_normalize_codice_fiscale(self, codice):
+        if re.match(r'^IT[0-9]{11}$', codice):
+            return codice[2:13]
+        return codice
+
+    @api.onchange('vat')
+    def _l10n_it_onchange_vat(self):
+        if not self.l10n_it_codice_fiscale:
+            self.l10n_it_codice_fiscale = self._l10n_it_normalize_codice_fiscale(self.vat)
+
+    @api.constrains('l10n_it_codice_fiscale')
+    def validate_codice_fiscale(self):
+        p = re.compile(r'^([A-Za-z]{6}[0-9]{2}[A-Za-z]{1}[0-9]{2}[A-Za-z]{1}[0-9]{3}[A-Za-z]{1}$)|([0-9]{11})')
+        for record in self:
+            if not p.match(record.l10n_it_codice_fiscale):
+                raise UserError(_("Invalid Codice Fiscale '%s': should be like 'MRTMTT91D08F205J' for physical person or '12345678901' for businesses (11 characters without 'IT')", record.l10n_it_codice_fiscale))
```

Comment:
Need to check that if you change something else on the partner it will not start complaining.  I wonder if we should not allow to add IT anyways...

Reply:
I was also wondering if that wasn't too restrictive. We can accept it since it is normalized to be used on the proxy. I changed that.